### ISSUE TITLE
staged_pps doesn't cope well with bisect_ppx -conditional

### ISSUE THREAD

**Issue by NathanReb** [ocaml/dune#1774]
I reported this issue on `bisect_ppx` [here](https://github.com/aantron/bisect_ppx/issues/185) first.

Basically what happened is that I needed to use both `ppx_import` and `bisect_ppx` on the same library resulting in a `preprocess` stanza looking like:
```
(preprocess
  (staged_pps
    ppx_import
    ...
    bisect_ppx -conditional
  )
)
```
This resulted in the library code always being instrumented. I assume the `-conditional` isn't properly passed somehow but I don't know much about the behaviour of `staged_pps`.

I suppose the fix for #57 will probably solve this issue but I can imagine a deeper preprocessor composition problem being involved here.

I'm as always happy to provide an example project and further detail as well as working on fix if needed!

**Comment by diml** [comment]
The syntax for passing flags to ppx rewriters is the following:

```
(staged_pps ppx_import ... bisect_ppx -- -conditional)
```

i.e. the flags are separated from the ppxs by a `--`. No using the `--` should result in an error as `dune` would end up looking for a `-conditional` ppx rewriter. This looks like a bug in dune.

Just to be sure, if you add the missing `--`, does the issue go away?

**Comment by diml** [comment]
Actually never mind,  I read the code again and `-conditional` should be recognised as a flag.

**Comment by diml** [comment]
Could you grep for `-ppx` in `_build/log` to see what `-ppx` command are being passed to the compiler?

**Comment by NathanReb** [comment]
I tried the adding the `--` just to check and the issue does persist even then.

It appears the command passed to `-ppx` is always just `ppx.exe --as-ppx`:
```
$ grep "\-ppx" _build/log 
$ (cd _build/default && /home/nathan/.opam/4.07.1/bin/ocamldep.opt -modules -ppx '.ppx/fd8acca92aaadae9d8099f94851fc84c/ppx.exe --as-ppx' -impl main.ml) > _build/default/.main.eobjs/main.ml.d
$ (cd _build/default && /home/nathan/.opam/4.07.1/bin/ocamlc.opt -w @a-4-29-40-41-42-44-45-48-58-59-60-40 -strict-sequence -strict-formats -short-paths -keep-locs -g -ppx '.ppx/fd8acca92aaadae9d8099f94851fc84c/ppx.exe --as-ppx' -bin-annot -I .main.eobjs -I /home/nathan/.opam/4.07.1/lib/bisect_ppx/runtime -no-alias-deps -opaque -o .main.eobjs/main.cmo -c -impl main.ml)
$ (cd _build/default && /home/nathan/.opam/4.07.1/bin/ocamlopt.opt -w @a-4-29-40-41-42-44-45-48-58-59-60-40 -strict-sequence -strict-formats -short-paths -keep-locs -g -ppx '.ppx/fd8acca92aaadae9d8099f94851fc84c/ppx.exe --as-ppx' -I .main.eobjs -I /home/nathan/.opam/4.07.1/lib/bisect_ppx/runtime -intf-suffix .ml -no-alias-deps -opaque -o .main.eobjs/main.cmx -c -impl main.ml)
```

This is from a small project with a hello world and the following dune file:
```
(executable
  (public_name main)
  (preprocess
    (staged_pps bisect_ppx -conditional)
  )
)
```

**Comment by diml** [comment]
Indeed, it was just an omission. Fixed in #1779
